### PR TITLE
 Pointer events don't transfer focus to the Flyout on Open (Issue #3154)

### PR DESCRIPTION
**Pull request by KAnder425**
In a previous fix, we set the AllowFocusOnInteraction property to false for the Flyout and its FlyoutPresenter to prevent some undesirable flyout dismissal problems.  See [https://github.com/microsoft/react-native-windows/pull/2817](https://github.com/microsoft/react-native-windows/pull/2817)
 
Unfortunately, that fix causes the focus to not transfer into any of the Flyout child elements when activated by a mouse/touch event.  The reason for this is that XAML makes a decision on whether or not to update the Focus element upon opening the flyout. If the action is due to a mouse/touch event and the AllowFocusOnInteraction property is set false on the Flyout, then the flyout doesn't update the Focus element. 
 
_(from %SDXROOT%\onecoreuap\windows\dxaml\xcp\components\focus\inc\FocusSelection.h)_
```
FocusSelection
    {
    public:
        template<class Element>
        static bool ShouldUpdateFocus(_In_ Element* const element, _In_ const DirectUI::FocusState focusState)
        {
            return !(focusState == DirectUI::FocusState::Pointer && GetAllowFocusOnInteraction(element) == false);
        }
 
```
_Called from FlyoutBase::OnPresenterLoaded (%SDXROOT%\onecoreuap\windows\dxaml\xcp\dxaml\lib\FlyoutBase_partial.cpp)._
 
This PR contains a refinement of that original flyout dismissal workaround that resolves the focus transfer problem. The change is to delay the setting of the AllowFocusOnInteraction flag to false until after the flyout is opened. 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3192)

### CODE REVIEW

**Comment by licanhua** [comment]
It's nice to have E2E test

**Comment by KAnder425** [comment]
I've opened a separate issue to add an e2e test for Flyouts.
https://github.com/microsoft/react-native-windows/issues/3256

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp [hunk 1]
```diff
@@ -182,17 +182,42 @@ void FlyoutShadowNode::createView() {
...
+            // translation based on an elevation derived from the count of
+            // open flyouts. We apply this translation on open of the flyout.
+            // (Translation is only supported on RS5+, eg. IUIElement9)
+            if (auto uiElement9 = GetView().try_as<winrt::IUIElement9>()) {
+              winrt::Numerics::float3 translation{
+                  0, 0, (float)16 * s_cOpenFlyouts};
+              flyoutPresenter.Translation(translation);
+            }
+
+            flyoutPresenter.AllowFocusOnInteraction(false);
```
Comment (khetanashita) [inline vnext/ReactUWP/Views/FlyoutViewManager.cpp]: Why do we need to set this again in here?

Comment (a00889920) [reply vnext/ReactUWP/Views/FlyoutViewManager.cpp]: +1 to this question

---
In reply to: [328731391](https://github.com/microsoft/react-native-windows/pull/3192#discussion_r328731391) [](ancestors = 328731391)

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp [hunk 2]
```diff
@@ -182,17 +182,42 @@ void FlyoutShadowNode::createView() {
...
+        if (!m_updating && instance != nullptr) {
+          if (auto flyoutPresenter = GetFlyoutPresenter()) {
+            // When multiple flyouts are overlapping, XAML's theme shadows
+            // don't render properly. As a workaround we enable a z-index
+            // translation based on an elevation derived from the count of
+            // open flyouts. We apply this translation on open of the flyout.
+            // (Translation is only supported on RS5+, eg. IUIElement9)
+            if (auto uiElement9 = GetView().try_as<winrt::IUIElement9>()) {
+              winrt::Numerics::float3 translation{
+                  0, 0, (float)16 * s_cOpenFlyouts};
```
Comment (a00889920) [inline vnext/ReactUWP/Views/FlyoutViewManager.cpp]: >0, 0 [](start = 18, length = 4)

[nit] params are missing comments of what they are

File: vnext/ReactUWP/Views/FlyoutViewManager.cpp (hunk 1 above)
Comment (KAnder425) [reply vnext/ReactUWP/Views/FlyoutViewManager.cpp]: In the earlier fix, setting AllowFocusOnInteraction false on the flyout prior to opening caused the associated flyoutpresenter to inherit that setting.  With this fix, that doesn't happen so we need to explicitly set it on the flyoutpresenter after opening.

---
In reply to: [328734469](https://github.com/microsoft/react-native-windows/pull/3192#discussion_r328734469) [](ancestors = 328734469,328731391)

**Pull request merged**